frontend/src/app/plot/plot.component.ts

```diff
@@ -153,6 +187,30 @@ export class PlotComponent {
     this.plot_data[index].visible = !this.plot_data[index].visible;
   }
 
+  /**
+   * Triggered when a datapoint is clicked on the plot, opens the
```

Comment:
Opens the option menu. Triggered....

1st line should always be what the function does. You can add conditions in future lines.